src/lib/parsers/include/sol-json.h

```diff
@@ -480,6 +480,181 @@ int sol_json_token_get_unescaped_string(const struct sol_json_token *token, stru
+ * @param value A pointer to the structure that will be filled with the value
+ *        of the element pointed by @a key_slice.
+ *
+ * @return If any parameter is invalid, or if @a scanner is not pointing to a
+ *         JSON Object, return -EINVAL. If @a key_slice is not present in the
+ *         JSON Object, -ENOENT. If @a key was found, and @a value was update
+ *         successfully with the value pointed by @a key, return 0.
+ *
+ * @see sol_json_array_get_at_index()
+ */
+int sol_json_object_get_key(struct sol_json_scanner *scanner, const struct sol_str_slice key_slice, struct sol_json_token *value);
+
+/**
+ * @brief Get the element in position @a i in JSON Array pointed by @a scanner
+ *
+ * @param scanner An initialized scanner, which is pointing to a JSON Array.
+ * @param value A pointer to the structure that will be filled with the value
+ *        of the element at position @a i.
+ * @param i The position of the desired element.
+ *
+ * @return If any parameter is invalid, or if @a scanner is not pointing to a
+ *         JSON Array, return -EINVAL. If @a i is larger than the array's
+ *         length, -ENOENT. If @a value was updated successfully with the value
+ *         in position @a i, return 0.
+ *
+ * @see sol_json_object_get_key()
+ */
+int sol_json_array_get_at_index(struct sol_json_scanner *scanner, uint16_t i, struct sol_json_token *value);
+
+/**
+ * @struct sol_json_path_scanner
+ *
+ * @brief Scanner used to go through segments of a JSON Path.
+ *
+ * @note JSONPath syntax is available at http://goessner.net/articles/JsonPath/.
+ *
+ * @see sol_json_path_scanner_init()
+ * @see sol_json_path_get_next_segment()
+ * @see SOL_JSON_PATH_FOREACH()
+ */
+struct sol_json_path_scanner {
+    const char *path; /**< @brief The JSONPath string. */
+    const char *end; /**< @brief Points to last character from path. */
+    /**
+     * @brief Points to last visited position from path and the beginning of
+     * next segment.
+     */
+    const char *current;
+};
+
+/**
+ * @brief Get the element pointed by the JSON Path @a path in @a scanner.
+ *
+ * @param scanner An initialized scanner, which is pointing to a JSON Object
+ *        or a JSON Array.
+ * @param path The JSON Path of the desired element.
+ * @param value A pointer to the structure that will be filled with the value
+ *        of the element pointed by @a path.
+ *
+ * @return If any parameter is invalid, or if @a scanner is not pointing to a
+ *         JSON Object or JSON Array, or if @a path is not a valid JSON Path,
+ *         return -EINVAL. If @a path is pointing to an invalid position in a
+ *         JSON Object or in a JSON Array, -ENOENT. If @a value was
+ *         successfully updated with the value of the element pointed by @a
+ *         path, return 0.
+ *
+ * @see sol_json_path_scanner
+ */
+int sol_json_get_path(struct sol_json_scanner *scanner, struct sol_str_slice path, struct sol_json_token *value);
```

Comment:
sol_json_object_get_key gets the child pointed by a key. sol_json_get_path uses a JSON Path instead of a key to look for the child.
Maybe the name is not clear enough. I'll rename it to sol_json_object_get_value_by_key and sol_json_get_value_by_path. Is it better?